### PR TITLE
Add config loader for json and yaml

### PR DESCRIPTION
**Pull request by darkdragon-001**
This adds config loaders for `json` and `yaml `formats in addition to `toml` format.

I am just not sure which packages to add to `setup.py`.

### CODE REVIEW

File: runrestic/runrestic/configuration.py [hunk 1]
```diff
@@ -122,7 +121,15 @@ def configuration_file_paths() -> Sequence[str]:
 def parse_configuration(config_filename: str) -> Dict[str, Any]:
     logger.debug(f"Parsing configuration file: {config_filename}")
     with open(config_filename) as file:
-        config = toml.load(file)
+        if config_filename.endswith('.json'):
+            import json
+            config = toml.load(file)
+        if config_filename.endswith(('yaml','.yml')):
+            import yaml
+            config = toml.load(file)
```
Comment (andreasnuesslein) [inline runrestic/runrestic/configuration.py:129]: that doesn't seem right. `toml.`

do you acutely need this? I mean it's nice and all but it does add extra maintenance for a problem I hadn't had this far.

**Comment by darkdragon-001** [comment]
> do you acutely need this?

This could be seen as preparation for #36. For a deeper hierarchy, I personally find `yaml` or `json` a lot clearer than `toml`'s `[key.sub.sub]` syntax.

> I mean it's nice and all but it does add extra maintenance for a problem I hadn't had this far.

Python's `json` and `yaml` libraries are used in many many applications and can be seen very stable. If these libraries break, then half of the world breaks. Therefore, most system administrators are more used to these formats (I had to search for toml syntax myself). Since it's only these few lines to maintain, I think this shouldn't add too much maintenance work. You can keep the schema and for the documentation documentation keep the toml examples (maybe just add a sentence telling json and yaml are also supported).

File: runrestic/runrestic/configuration.py (hunk 1 above)
Comment (darkdragon-001) [reply runrestic/runrestic/configuration.py]: I updated the commit.

**Comment by andreasnuesslein** [comment]
> Python's json and yaml libraries are used in many many applications and can be seen very stable. If these libraries break, then half of the world breaks.

That's not what I mean. I'm just talking about a general overhead that is introduced by a feature nobody actually needs (it seems).

However I'm also not saying I won't merge it - but could you please add some tests so that when it breaks the tests might let us know about it first? And update the docs a bit to let people know they could also use yaml or json directly

Cheers

**Comment by darkdragon-001** [comment]
> I'm just talking about a general overhead that is introduced by a feature nobody actually needs (it seems).

As long as something can be achieved _somehow_, nothing is _needed_, still could be _preferred_. Furthermore, how do you come to the conclusion that _nobody_ would like to have such a feature?

> However I'm also not saying I won't merge it - but could you please add some tests so that when it breaks the tests might let us know about it first? And update the docs a bit to let people know they could also use yaml or json directly

Okay, I will look for the right places, to add the docs and tests...

**Pull request closed without merging**